src/interface/lazy.jl

```diff
@@ -29,6 +29,7 @@ function LazyTensor{T}(arr::Base.AbstractArrayOrBroadcasted) where {T}
     LazyTensor{eltype(arr), ndims(arr)}(tns, extrude)
 end
 LazyTensor(arr::Tensor) = LazyTensor{eltype(arr)}(arr)
+LazyTensor(swizzle_arr::SwizzleArray) = permutedims(LazyTensor(swizzle_arr.body), typeof(swizzle_arr).parameters[1])
```

Comment:
this might be more cleanly written with a `where` statement, i.e. `LazyTensor(swizzle_arr::SwizzleArray{Tns, dims}) where {Tns, dims} =`

Reply:
Done!